sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -281,6 +281,34 @@ export class ManagementClient extends LinkEntity {
       : undefined;
   }
 
+  /**
+   * Evaluates the `associatedLinkName` to use for given Service Bus entity client instance
+   * @param forSender Boolean flag indicating if this for an operation on sender or receiver
+   * @param clientEntityContext The `ClientEntityContext` associated with given Service Bus entity client
+   * @param sessionId `sessionId` if applicable
+   */
+  private _getAssociatedLinkName(
```

Comment:
I would skip the sender part altogether and have the helper method handle only the receiver cases.
In case of sender, it is very straightforward to get the associated link name, we pick it from the `clientEntityContext.sender`.
Its the case of receiver which is not straightforward.

- Remove sender related parts & rename helper method to `_getReceiverAssociatedLinkName()`
- Update the scehduledMessages & cancelScheduledMessages calls to use the `clientEntityContext.sender.name` directly.